core/include/scipp/core/transform_common.h

```diff
@@ -56,6 +60,18 @@ static constexpr auto dimensionless_unit_check_return =
       return aUnit;
     };
 
+namespace transform_flags {
+/// Add this to overloaded operator to indicate that the operation does not
+/// produce output with variances, even if the inputs contain variances.
+static constexpr auto no_variance_output = []() {};
+using no_variance_output_t = decltype(no_variance_output);
+
+template <int N> static constexpr auto expect_no_variance_arg = []() {};
+template <int N>
+using expect_no_variance_arg_t = decltype(expect_no_variance_arg<N>);
```

Comment:
Would this work: 
```cpp
static constexpr auto dummy_lambda = []() {};
using dummy_lambda_t = decltype(dummy_lambda);
using no_variance_output_t = dummy_lambda_t;
template <int N>
using expect_no_variance_arg_t = dummy_lambda_t;
```

Reply:
No it would not work, since `transform` checks for the flag based on its type, so `N` must be part of the type. Having it part of the typedef is not sufficient.